xbmc/addons/gui/skin/SkinTimerManager.cpp

```diff
@@ -96,38 +96,32 @@ void CSkinTimerManager::LoadTimerInternal(const TiXmlElement* node)
         node->FirstChild("stop")->FirstChild()->ValueStr());
   }
 
-  if (node->FirstChild("onstart") && node->FirstChild("onstart")->FirstChild() &&
-      !node->FirstChild("onstart")->FirstChild()->ValueStr().empty())
+  // process onstart actions
+  startActions.EnableSendThreadMessageMode();
+  const TiXmlElement* onStartElement = node->FirstChildElement("onstart");
+  while (onStartElement)
```

Comment:
I would write this a bit different, because if  `onStartElement` is `nullptr`, you don't even need to construct `startActions`.

```c++
const TiXmlElement* onStartElement = node->FirstChildElement("onstart");
if (onStartElement)
{
  CGUIAction startActions;
  startActions.EnableSendThreadMessageMode();
  while (onStartElement)
  {
  ...
```

Same applies for the code handling "onstop" below.

Reply:
Ah, okay.